detectable/src/main/java/com/synopsys/integration/detectable/detectables/projectinspector/ProjectInspectorParser.java

```diff
@@ -84,37 +86,41 @@ public CodeLocation codeLocationFromModule(ProjectInspectorModule module) {
         Map<String, Dependency> lookup = new HashMap<>();
 
         //build the map of all external ids
-        module.dependencies.forEach(dependency -> lookup.computeIfAbsent(dependency.id, missingId -> convertProjectInspectorDependency(dependency)));
+        module.components.forEach((dependencyId, component) -> {
+            lookup.computeIfAbsent(dependencyId, missingId -> convertProjectInspectorDependency(component));
+        });
 
         //and add them to the graph
         DependencyGraph mutableDependencyGraph = new BasicDependencyGraph();
-        module.dependencies.forEach(moduleDependency -> {
-            Dependency dependency = lookup.get(moduleDependency.id);
-            moduleDependency.includedBy.forEach(parent -> {
-                if ("DIRECT".equals(parent)) {
-                    mutableDependencyGraph.addDirectDependency(dependency);
-                } else if (lookup.containsKey(parent)) {
-                    mutableDependencyGraph.addChildWithParent(dependency, lookup.get(parent));
-                } else { //Theoretically should not happen according to PI devs. -jp
-                    throw new RuntimeException("An error occurred reading the project inspector output." +
-                        " An unknown parent dependency was encountered '" + parent + "' while including dependency '" + moduleDependency.name + "'.");
-                }
-            });
+        module.components.forEach((moduleDependencyId, moduleDependency) -> {
+            Dependency dependency = lookup.get(moduleDependencyId);
+            if(moduleDependency.inclusionType.equals("DIRECT")){
+                mutableDependencyGraph.addDirectDependency(dependency);
+            } else if (moduleDependency.inclusionType.equals("TRANSITIVE")) {
```

Comment:
I have seen some Detectors marking dependencies as both direct and transitive.

From the PI docs on `IncludedBy`:

> Includes a reference to every dependency that transitively requires this  dependency.

I wonder if this field is ever populated when a dependency is both direct and transitive. And if so, I wonder if it would be valuable to pass that data along to Detect results.

Reply:
Good Point. I ran this version of PI for multiple projects and I did see some examples which have this kind of dependencies. The output of PI in that case would have two objects of same kind, one for DIRECT and another for TRANSITIVE, so in essence this is being already taken care of. If thats what you were implying.

Reply:
Thanks for the clarification. Yeah, that's exactly what I was wondering about.